### PR TITLE
NEW - List product in orders

### PR DESCRIPTION
**Pull request by BB2A-Anthony**
# NEW|New [*List product in orders*]

The goal is to have a list of products on order and to be able to generate supplier orders in the future.

### CODE REVIEW

**Comment by BB2A-Anthony** [comment]
@eldy Your opinion on the subject?

**Comment by eldy** [comment]
So this means the page will no more be the "list of orders" but the "lines of orders".
i think it is better to have another menu entry dedicated to this for example under the "List" menu entry, we can have "Lines of orders" (added if we are in development mode MAIN_FEATURES_LEVEL >= 2 to start)
The default fields visible, filters, and more will probably not be the same for the 2 needs, so may be a dedicated page will be better instead of a hack of existing page.

File: htdocs/commande/list_line.php [hunk 1]
```diff
@@ -0,0 +1,2660 @@
...
+
+		// Action column
+		if (empty($conf->global->MAIN_CHECKBOX_LEFT_COLUMN)) {
+			print '<td class="nowrap center">';
+			if ($massactionbutton || $massaction) {   // If we are in select mode (massactionbutton defined) or if we have already selected and sent an action ($massaction) defined
+				$selected = 0;
+				if (in_array($obj->rowid, $arrayofselected)) {
+					$selected = 1;
+				}
+				print '<input id="cb'.$obj->rowid.'" class="flat checkforselect" type="checkbox" name="toselect[]" value="'.$obj->rowid.'"'.($selected ? ' checked="checked"' : '').'>';
```
Comment (fappels) [inline htdocs/commande/list_line.php:2600]: Should be line id to have mass actions on line level.

Comment (bb2a) [reply htdocs/commande/list_line.php:2585]: $obj->rowid it's line id what is the problem?

**Comment by bb2a** [comment]
> 

@eldy i have execut your suggest it's ok for you?

**Comment by bb2a** [comment]
@eldy the work is goog?

**Comment by BB2A-Anthony** [comment]
@eldy anything else to modify for validation?

**Comment by BB2A-Anthony** [comment]
@eldy it's not ready to merge?

File: htdocs/commande/list_det.php [hunk 2]
```diff
@@ -0,0 +1,2168 @@
...
+		// Show shippable Icon (this creates subloops, so may be slow)
+		if (!empty($arrayfields['shippable']['checked'])) {
+			print '<td class="center">';
+			if (!empty($show_shippable_command) && !empty($conf->stock->enabled)) {
+				if (($obj->fk_statut > $generic_commande::STATUS_DRAFT) && ($obj->fk_statut < $generic_commande::STATUS_CLOSED)) {
+					$generic_commande->getLinesArray(); 	// Load array ->lines
+					$generic_commande->loadExpeditions();	// Load array ->expeditions
+
+					$numlines = count($generic_commande->lines); // Loop on each line of order
+					for ($lig = 0; $lig < $numlines; $lig++) {
```
Comment (fappels) [inline htdocs/commande/list_det.php:1980]: Better to get shippable status for line, so you know which line is shippable (no need to loop order lines)

File: htdocs/langs/en_US/orders.lang [hunk 3]
```diff
@@ -96,6 +96,10 @@ OrdersStatisticsSuppliers=Purchase order statistics
 NumberOfOrdersByMonth=Number of orders by month
 AmountOfOrdersByMonthHT=Amount of orders by month (excl. tax)
 ListOfOrders=List of orders
+ListOrderLigne=Lines of orders
+productobuy=Products to buy only
+productonly=Products only
+disablelinefree=No lines free
```
Comment (fappels) [inline htdocs/langs/en_US/orders.lang:102]: No free lines

**Pull request merged**